### PR TITLE
Add hqc

### PR DESCRIPTION
**Pull request by TomerHawk**

### CODE REVIEW

File: encryption/hqc/hqc-rmrs/impl/avx2.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+name: AVX2
+platform: 'AVX2 , C'
```
Comment (dingens) [inline encryption/hqc/hqc-rmrs/impl/avx2.yaml:2]: please use `platform: c` and `hardware requirements: [ avx2 ]`

File: encryption/hqc/hqc-rmrs/impl/ref.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+name: Reference
+platform: 'Intel Core i7-7820X CPU, 3.6GHz '
+type: reference
+hardware features:
+  - 'Hyper-Threading, Turbo Boost and SpeedStep features were disabled '
```
Comment (dingens) [inline encryption/hqc/hqc-rmrs/impl/ref.yaml:5]: this is about which hardware features are required (for example, hardware aes acceleration or vector instructions). probably `[]` for the reference impl.

File: encryption/hqc/hqc/impl/aarch64.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+name: aarch64
+sources:
+  - 'https://bench.cr.yp.to/results-kem.html'
+platform: 'aarch64, rpi4ubuntu64, armv8, 4 x 1500MHz'
+hardware features:
+  - '4 x 1500MHz, A72 (410fd083); 2019 Broadcom BCM2711'
```
Comment (dingens) [inline encryption/hqc/hqc/impl/aarch64.yaml:6]: please note that `impl` and `bench` are different things.

`implementation` is just about static code, things like cpu speed are properties of the `benchmark`

`impl.platform` is just something like `c`, `assembler` or `fpga` (I agree the name clash is unfortunate).

about the hardware features: see above

File: encryption/hqc/hqc/impl/avx2.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+name: AVX2
+type: optimized
+platform: 'AVX2, in C'
```
Comment (dingens) [inline encryption/hqc/hqc/impl/avx2.yaml:3]: dito

File: encryption/hqc/hqc/impl/ref.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+name: Reference
+platform: 'Intel Core i7-7820X CPU, 3.6GHZ'
+type: reference
+hardware features:
+  - 'Hyper-Threading, Turbo Boost and SpeedStep features were disabled'
```
Comment (dingens) [inline encryption/hqc/hqc/impl/ref.yaml:5]: dito

File: encryption/hqc/hqcpke/impl/aarch64.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+name: aarch64
+sources:
+  - 'https://bench.cr.yp.to/results-kem.html'
+platform: 'aarch64, rpi4ubuntu64, armv8, 4 x 1500MHz'
+hardware features:
+  - '4 x 1500MHz, A72 (410fd083); 2019 Broadcom BCM2711'
```
Comment (dingens) [inline encryption/hqc/hqcpke/impl/aarch64.yaml:6]: dito

File: encryption/hqc/hqcpke/impl/avx2.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+name: AVX2
+type: optimized
+platform: 'AVX2, in C'
```
Comment (dingens) [inline encryption/hqc/hqcpke/impl/avx2.yaml:3]: dito

File: encryption/hqc/hqcpke/impl/ref.yaml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+name: Reference
+platform: 'Intel Core i7-7820X CPU, 3.6GHZ'
+type: reference
+hardware features:
+  - 'Hyper-Threading, Turbo Boost and SpeedStep features were disabled'
```
Comment (dingens) [inline encryption/hqc/hqcpke/impl/ref.yaml:5]: dito

File: encryption/hqc/hqc/hqc.yaml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+name: HQC
```
[the remark this answers is not in the document]
Comment (TomerHawk) [reply encryption/hqc/hqc/hqc.yaml:1]: Sure, I will fix them.

**Pull request merged**